pollux/lib/core/src/main/scala/io/iohk/atala/pollux/core/service/PresentationServiceImpl.scala

```diff
@@ -91,15 +100,55 @@ private class PresentationServiceImpl(
         )
       )
 
-      presentationPayload <- createPresentationPayloadFromCredential(
+      presentationPayload <- createJwtPresentationPayloadFromCredential(
         issuedCredentials,
-        record.credentialFormat,
         requestPresentation,
         prover
       )
     } yield presentationPayload
   }
 
+  override def createAnoncredPresentationPayloadFromRecord(
+      recordId: DidCommID,
+      prover: Issuer,
+      issuanceDate: Instant
+  ): ZIO[WalletAccessContext, PresentationError, AnoncredPresentation] = {
+
+    for {
+      maybeRecord <- presentationRepository
+        .getPresentationRecord(recordId)
+        .mapError(RepositoryError.apply)
```

Comment:
I think, it's a good practice to call `logError` as well before `mapError`, so we can track what exactly is wrong

Reply:
The best way to handle this should defined in the ADR part of the `Error Management` value brief. I'll add this specific case - call to `mapError()` - to the list 👍

Reply:
Noted